rzls: init at 9.0.0-preview.25052.3

tris203 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
This PR introduces a derivation for building rzls (the Razor Language Server) from the dotnet/razor repository. While the C# plugin for VS Code bundles rzls for use within the editor, this derivation makes it accessible for other editors by building it separately.

Currently, the versioning process is manual due to Microsoft's build practices. There is an open issue in their repo about improving version tags to enable automated updates in the future.

Fix #369979

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

tris203 committed:
rzls: init at 9.0.0-preview.25052.3

tris203 committed:
maintainers: add tris203

tris203 commented:
duplicate of #371569 but with a slightly newer version
@bretek do you have a preference over which one we go with? seems we came up with almost exactly the same thing

tris203 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 371923`

tris203 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 371923`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rzls</li>
  </ul>
</details>

l1npengtul commented on pkgs/by-name/rz/rzls/package.nix:
```diff
@@ -0,0 +1,98 @@
+
+  meta = {
+    homepage = "https://github.com/dotnet/vscode-csharp";
+    description = ''
```
Descriptions should not be complete sentences and should be phrases that fill the blank in "<package> is a/an/the ____"

tris203 replied:
where is that documentation? I have since found this, but that doesn't seem to agree, but I want to make sure that I get the wording right

https://github.com/NixOS/nixpkgs/blob/73a667ff6a0326f8cfb99206415bab577a5dcbf5/pkgs/README.md?plain=1#L401C2-L411C5

l1npengtul replied:
Please see https://nixos.org/manual/nixpkgs/stable/#var-meta-description

In particular,
- "Not end with a period (or any punctuation for that matter)."
- "Be short, just one sentence."

tris203 commented on pkgs/by-name/rz/rzls/package.nix:
```diff
@@ -0,0 +1,98 @@
+  meta = {
+    homepage = "https://github.com/dotnet/vscode-csharp";
+    description = ''
+      Razor is a markup syntax for adding server-side logic to web pages. This package
+            contains a Razor language server.'';
```
```suggestion
      Language server for Razor markup syntax to add C# server-side logic to web pages'';
```

Would this be better?

l1npengtul replied:
This is good.

bretek commented:
> duplicate of #371569 but with a slightly newer version @bretek do you have a preference over which one we go with? seems we came up with almost exactly the same thing

@tris203 😄 they do look very similar. I prefer mine because I've removed a lot of the flags and comments that seem to be unnecessary / specific to roslyn-ls. I've also now added an update script and brought my pull request up to 25052.3

tris203 commented:
> > duplicate of #371569 but with a slightly newer version @bretek do you have a preference over which one we go with? seems we came up with almost exactly the same thing
> 
> @tris203 😄 they do look very similar. I prefer mine because I've removed a lot of the flags and comments that seem to be unnecessary / specific to roslyn-ls. I've also now added an update script and brought my pull request up to 25052.3

i agree, yours is cleaner. especially with the update script @bretek. would you be willing to cherry pick my maintainer commit onto your branch, and me as a maintainer in your derivation?

tris203 commented:
superceeded by https://github.com/NixOS/nixpkgs/pull/371569
